editoast/src/views/infra/mod.rs

```diff
@@ -586,20 +597,57 @@ pub mod tests {
         let infra: Infra =
             call_and_read_body_json(&app, create_infra_request("get_voltages_test")).await;
 
+        let test_cases = vec![true, false];
+        // Create catenary
         let catenary = Catenary {
             id: "test".into(),
             voltage: "0".into(),
             track_ranges: vec![],
         };
+
         let req = create_object_request(infra.id.unwrap(), catenary.into());
         assert_eq!(call_service(&app, req).await.status(), StatusCode::OK);
 
-        let req = TestRequest::get()
-            .uri(format!("/infra/{}/voltages/", infra.id.unwrap()).as_str())
-            .to_request();
-        let response = call_service(&app, req).await;
-        assert_eq!(response.status(), StatusCode::OK);
-
+        // Create rolling_stock
+        let rolling_stock: RollingStockModel = get_other_rolling_stock();
+        let post_response = call_service(
+            &app,
+            TestRequest::post()
+                .uri("/rolling_stock")
+                .set_json(rolling_stock)
+                .to_request(),
+        )
+        .await;
+        let rolling_stock: RollingStockModel = read_body_json(post_response).await;
+        let rolling_stock_id = rolling_stock.id.unwrap();
+
+        for include_rolling_stock_modes in test_cases {
+            let req = TestRequest::get()
+                .uri(
+                    format!(
+                        "/infra/{}/voltages/?include_rolling_stock_modes={}",
+                        infra.id.unwrap(),
+                        include_rolling_stock_modes
+                    )
+                    .as_str(),
+                )
+                .to_request();
+            let response = call_service(&app, req).await;
+            assert_eq!(response.status(), StatusCode::OK);
+
+            if !include_rolling_stock_modes {
+                let voltages: Vec<String> = read_body_json(response).await;
+                assert_eq!(voltages[0], "0");
+                assert_eq!(voltages.len(), 1);
+            } else {
+                let voltages: Vec<String> = read_body_json(response).await;
+                assert_eq!(voltages[1], "25000");
+            }
+        }
+        // Delete Rolling_stock
+        let delete_request = rolling_stock_delete_request(rolling_stock_id);
+        let delete_response = call_service(&app, delete_request).await;
+        assert_eq!(delete_response.status(), StatusCode::NO_CONTENT);
```

Comment:
Can you use a fixture instead of creating a new RS and a new infra ? :) This way, no need to remove them at the end of the test (and if the test fails, the db will not be polluted)

Reply:
will be resolved by #4157